File: app/src/main/java/com/example/myapplication/api/models/AppointmentDTO.java
package com.example.myapplication.api.models;

public class AppointmentDTO {
    private String id;
    private String patientId;
    private String doctorId;
    private String appointmentDate;
    private String reason;
    private String notes;
    private String doctorNotes;
    private String status;
    private String patientName;
    private String doctorName;
    private String doctorSpecialization;

    public String getId() { return id; }
    public String getPatientId() { return patientId; }
    public String getDoctorId() { return doctorId; }
    public String getAppointmentDate() { return appointmentDate; }
    public String getReason() { return reason; }
    public String getNotes() { return notes; }
    public String getDoctorNotes() { return doctorNotes; }
    public String getStatus() { return status; }
    public String getPatientName() { return patientName; }
    public String getDoctorName() { return doctorName; }
    public String getDoctorSpecialization() { return doctorSpecialization; }
}

File: ClinicAPI/src/API/DTOs/AppointmentDTO.cs
using API.Models;

namespace API.DTOs;

public class AppointmentDTO
{
    public string? Id { get; set; }
    public string PatientId { get; set; } = string.Empty;
    public string DoctorId { get; set; } = string.Empty;
    public DateTime AppointmentDate { get; set; }
    public string? Reason { get; set; }
    public string? Notes { get; set; }
    public string? DoctorNotes { get; set; }
    public string Status { get; set; } = "Scheduled";

    // Additional info for display
    public string? PatientName { get; set; }
    public string? DoctorName { get; set; }
    public string? DoctorSpecialization { get; set; }
}

public class CreateAppointmentRequest
{
    public string PatientId { get; set; } = string.Empty;
    public string DoctorId { get; set; } = string.Empty;
    public DateTime AppointmentDate { get; set; }
    public string? Reason { get; set; }
    public string? Notes { get; set; }
}

public class CompleteAppointmentRequest
{
    public string? DoctorNotes { get; set; }
}
